### PR TITLE
Add additional trigger for policy table update

### PR DESCRIPTION
**Pull request by LevchenkoS**
According to the new requirements the policy table update
should be trigered in case of `navi` application has been registered
and `module_config` has no `certificate`.
##
- `policy_manager.h` moved to `components/include`;
- `mock_policy_manager.h` deleted as a duplicate;
- `mock_app_stopwatch.h` deleted as a duplicate
##
Fixed coding style
##
@LuxoftAKutsan, @dtrunov, @AGaliuzov, @okozlovlux, please, review.

### CODE REVIEW

File: src/components/policy/policy_regular/test/policy_manager_impl_test.cc [hunk 1]
```diff
@@ -675,6 +675,19 @@ TEST_F(PolicyManagerImplTest, AddAppStopwatch) {
   manager->Add("12345", usage_statistics::SECONDS_HMI_FULL, 30);
 }
 
+TEST_F(PolicyManagerImplTest,
+       TiggerPTUForNaviAppInCaseNoCertificateExistsInPolicyTable) {
```
Comment (okozlovlux) [inline src/components/policy/policy_regular/test/policy_manager_impl_test.cc]: @LevchenkoS , can provide some expected result in test name?

Comment (LevchenkoS) [inline src/components/policy/policy_regular/test/policy_manager_impl_test.cc]: @okozlovlux, fixed in 7e37bb3

**Comment by LevchenkoS** [comment]
Rebuild required

**Comment by LuxoftAKutsan** [comment]
@LevchenkoS are you sure that this PR should be merged to develop?

File: src/components/policy/policy_regular/include/policy/cache_manager.h [hunk 2]
```diff
@@ -215,6 +215,15 @@ class CacheManager : public CacheManagerInterface {
       std::map<std::string, StringArray>& app_hmi_types);
 
   /**
+   * @brief AppHasHMIType checks whether app has been registered with certain
+   *HMI type.
+   *
+   * @return true in case app contains certain HMI type, false otherwise.
+   */
+  virtual bool AppHasHMIType(const std::string& application_id,
```
Comment (dtrunov) [inline src/components/policy/policy_regular/include/policy/cache_manager.h]: @LevchenkoS please use override

File: src/components/policy/policy_regular/include/policy/cache_manager.h [hunk 3]
```diff
@@ -215,6 +215,15 @@ class CacheManager : public CacheManagerInterface {
...
+                             policy_table::AppHMIType hmi_type) const;
+
```
Comment (dtrunov) [inline src/components/policy/policy_regular/include/policy/cache_manager.h]: @LevchenkoS Please use OVERRIDE

Comment (AGaliuzov) [inline src/components/policy/policy_regular/include/policy/cache_manager.h]: @LevchenkoS even better to use `FINAL` here

Comment (LevchenkoS) [inline src/components/policy/policy_regular/include/policy/cache_manager.h]: @dtrunov, @AGaliuzov, fixed in 8a9a498

**Pull request merged**